components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/handler/impl/BasicAuthenticationHandler.java

```diff
@@ -58,48 +58,48 @@ public void init(InitConfig initConfig) {
 
     @Override
     public String getName() {
+
         return "BasicAuthentication";
     }
 
     @Override
     public int getPriority(MessageContext messageContext) {
+
         return getPriority(messageContext, 100);
     }
 
     @Override
     public boolean canHandle(MessageContext messageContext) {
-        if ( messageContext instanceof AuthenticationContext ) {
+
+        if (messageContext instanceof AuthenticationContext) {
             AuthenticationContext authenticationContext = (AuthenticationContext) messageContext;
             if (authenticationContext.getAuthenticationRequest() != null) {
                 String authorizationHeader = authenticationContext.getAuthenticationRequest().
                         getHeader(HttpHeaders.AUTHORIZATION);
-                if ( StringUtils.isNotEmpty(authorizationHeader) && authorizationHeader.startsWith(BASIC_AUTH_HEADER)
-                        ) {
-                    return true;
-                }
+                return StringUtils.isNotEmpty(authorizationHeader) && authorizationHeader.startsWith(BASIC_AUTH_HEADER);
```

Comment:
This would allow a user to authenticate with
Basic username:password
header

Would be better to either check whether it starts with "Basic " or split the string and check whether its exactly matches "Basic"

Reply:
It does have ```&& authorizationHeader.startsWith(BASIC_AUTH_HEADER)```